modules/core/src/lifecycle/component-state.js

```diff
@@ -175,6 +175,9 @@ export default class ComponentState {
 
   // Set normal, non-async value
   _setPropValue(propName, value) {
+    // Save the current value before overwriting so that diffProps can access both
+    this.freezeAsyncOldProps();
```

Comment:
Hopefully we don't have too much perf impact by always doing this.

Reply:
It is only done if an async prop is changed.